src/libsyntax/parse/parser.rs

```diff
@@ -2847,7 +2847,11 @@ impl<'a> Parser<'a> {
             // want to keep their span info to improve diagnostics in these cases in a later stage.
             (true, Some(AssocOp::Multiply)) | // `{ 42 } *foo = bar;` or `{ 42 } * 3`
             (true, Some(AssocOp::Subtract)) | // `{ 42 } -5`
```

Comment:
If these are included, why isn't `{ 42 } & 3`? Surely the have the same property?

Reply:
Ah this seems like a reporting-only heuristic and only the `can_continue_expr_unambiguously` check is relevant to what I was seeing.